webpack/assets/javascripts/react_app/components/users/PersonalAccessTokens/PersonalAccessTokens.js

```diff
@@ -0,0 +1,69 @@
+import React, { Fragment, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import NewPersonalAccessToken from './NewPersonalAccessToken';
+import PersonalAccessTokenForm from './PersonalAccessTokenForm';
+import PersonalAccessTokensList from './PersonalAccessTokensList';
+import { translate as __ } from '../../../common/I18n';
+
+const PersonalAccessTokens = ({
+  url,
+  canCreate,
+  tokens,
+  newPersonalAccessToken,
+  clearNewPersonalAccessToken,
+  getPersonalAccessTokens: getPersonalAccessTokensAction,
+  revokePersonalAccessToken: revokePersonalAccessTokenAction,
+}) => {
+  const getPersonalAccessTokens = useCallback(
+    () => getPersonalAccessTokensAction({ url }),
+    [getPersonalAccessTokensAction, url]
+  );
+
+  const revokePersonalAccessToken = id =>
+    revokePersonalAccessTokenAction({ url, id });
+
+  useEffect(() => {
+    getPersonalAccessTokens();
+  }, [getPersonalAccessTokens]);
```

Comment:
Even though it may avoid the eslint warning, I think this use of `useEffect` still breaks [this rule](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) because of the `useCallback` inside of `getPersonalAccessTokens`.

I'm not sure why `useCallback` would be needed here.  If you just need to ensure that the API call only occurs on mount, you can pass an empty dependency array:
```suggestion
  useEffect(() => {
    getPersonalAccessTokensAction({ url });
  }, []);
```